howso/typing.amlg

```diff
@@ -147,53 +539,427 @@
+                                        "is not specified."
+                                    )
+                            )
+                        feature_residuals_full
+							(assoc
+								type  "boolean"
+								description
+                                    (concat
+                                        "If True, outputs feature residuals for all (context and action) "
+                                        "features locally around the prediction. Uses only the context "
+                                        "features of the reacted case to determine that area. Uses "
+                                        "full calculations, which uses leave-one-out for cases for computations."
+                                    )
+							)
+                        feature_residuals_robust
+							(assoc
+								type  "boolean"
+								description
+                                    (concat
+                                        "If True, outputs feature residuals for all (context and action) "
+                                        "features locally around the prediction. Uses only the context "
+                                        "features of the reacted case to determine that area. Uses robust "
+                                        "calculations, which uses uniform sampling from the power set of "
+                                        "features as the contexts for predictions. 'selected_prediction_stats' "
+                                        "controls the returned prediction stats."
+                                    )
+							)
+                        prediction_stats
+							(assoc
+								type  "boolean"
+								description
+                                    (concat
+                                        "When true outputs feature prediction stats for all (context "
+                                        "and action) features locally around the prediction. The stats "
+                                        "returned  are (\"r2\", \"rmse\", \"spearman_coeff\", \"precision\", "
+                                        "\"recall\", \"accuracy\", \"mcc\", \"confusion_matrix\", \"missing_value_accuracy\"). "
+                                        "Uses only the context features of the reacted case to determine that area. "
+                                        "Uses full calculations, which uses leave-one-out context features for "
+                                        "computations. 'selected_prediction_stats' controls the returned prediction stats."
+                                    )
+							)
+                        selected_prediction_stats
+                            (assoc
+                                type "list"
+                                values
+                                    (assoc
+                                        type "string"
+                                        enum (get_value !supportedPredictionStats)
```

Comment:
Could this enum be a schema instead? Might be easier to use from a typing perspective.

Reply:
Also looks like it might be used in react aggregate? Should probably reference it there too, currently that is just a freeform string.